src/storage/interpreter.rs

```diff
@@ -137,6 +137,34 @@ pub trait InterpreterStorage:
         self.storage::<ContractsState>().remove(&(contract, key))
     }
 
+    /// Fetch a range of values from a key-value mapping in a contract storage.
+    /// Returns the full range requested using optional values in case
+    /// a requested slot is unset.  
+    fn merkle_contract_state_range(
```

Comment:
Hmm, do we want to duplicate each method with "ranged" version? Maybe better to add `range` to `merkle_contract_state` and don't use it right now=)

Reply:
Not all storage ops are going to be ranged, for example SRW only fetches a single word instead of an entire slot. So it doesn't take a range.

Reply:
Does `merkle_contract_state_range` with `range = 0` returns the same result as `merkle_contract_state`?

Reply:
hmm according to the [specs](https://github.com/FuelLabs/fuel-specs/blob/master/specs/vm/instruction_set.md#srwq-state-read-sequential-32-byte-slots), it would return and do nothing.

Reply:
Oh, sorry, I meant with `range = 1`=)

Reply:
Yeah it would be the same result as before in that case

Reply:
So, do we want to have only one method and use it with `range = 1`, or do we prefer two separate methods?

Reply:
I figured its more efficient to have separate methods, so that we don't have do deal with heap types more than we have to

Reply:
If `Storage` returns only `Set` slots, we can simplify `merkle_contract_state_range` return type into `Result<Vec<Cow<Bytes32>>, Self::DataError>;`, and the processing for `range = 1` will be the same as in `merkle_contract_state`.

But yea, we still will work with heap type - `Vec` so let's keep both methods=)